packages/core/codeframe/src/codeframe.js

```diff
@@ -46,6 +46,9 @@ export default function codeFrame(
 ): string {
   if (highlights.length < 1) return '';
 
+  // Replace all tabs with a space to simplify charcount
+  code = code.replace(/\t/g, ' ');
```

Comment:
Could you add a comment here that tabs should be treated as a single space
 https://github.com/parcel-bundler/parcel/blob/5e38099af6d4cf53d5d6e411b11bbd706ffc7434/packages/core/diagnostic/src/diagnostic.js#L10-L15

Are you sure this is what babel does (ignore the wrong codehighligh locations for now)? The indentation is barely noticeable:

![Bildschirmfoto 2019-11-01 um 15 25 41](https://user-images.githubusercontent.com/4586894/68031416-e7d39d80-fcbb-11e9-9477-c886084eda0f.png)

Reply:
I'm not sure this is what babel does, except for the tab space at the start they don't really seem to do anything with tabs

Reply:
In any case we should count a tab as 1 character length as it can probably be different from machine to machine

Reply:
`@babel/cli` displays tabs with the width of two spaces, but it really is a tab character.
Printing a tab with  `printf "x\tx"` has a width of 8. 🤷‍♂

Reply:
> In any case we should count a tab as 1 character length as it can probably be different from machine to machine

But then the highlight column would have to be adjusted as well?

Reply:
I'll make the tab width variable (through a param or const in the codeframe-util no fancy editorconfig [yet])

But it will still be just length 1 for the highlights. I think it will be the most reliable way to do this as tabs don't really have a fixed length and are just 1 character after all

Reply:
I could also just keep tabs and not replace them with spaces if that seems better?

Reply:
> But a tab will still just count as 1 character for the highlights. I think it will be the most reliable way to do this as tabs don't really have a fixed length and are just 1 character after all

So the one who prints the codeframe is responsible for expanding these counts as necessary? 👍 

(I don't really care about the exact tab width, just that the source line and highlight arrows align 😄 )

Reply:
> But then the highlight column would have to be adjusted as well?

how come, in that test the `\t` still counts as a single character, they apparently just don't replace it with spaces

Reply:
> So the one who prints the codeframe is responsible for expanding these counts as necessary? 👍

Exactly, the codeframe util will have to figure it out, or whatever new formatters we come up with in the future

Reply:
I think this would work: count highlight column in chars (tab = 1), then when it is printed, then for the highlight line:
```js
sourceLine.splice(0, column).replace(/\s/g, " ")/*either keep tabs here or replace with 2 spaces*/
   + "^^^^^" + highlight.message
```

Reply:
Yeah seems like that's the only real way of doing it, ours is slightly more complicated though due to supporting multiple highlights